cmd/release-controller/sync.go

```diff
@@ -475,6 +478,22 @@ func (c *Controller) syncAccepted(release *Release) error {
 	return nil
 }
 
+func (c *Controller) syncTests(release *Release) error {
+	testTags := findTagReferencesByPhase(release, releasePhaseAccepted, releasePhaseRejected)
+	if glog.V(4) && len(testTags) > 0 {
+		glog.Infof("release=%s validation=%v", release.Config.Name, tagNames(testTags))
+	}
+	for _, tag := range testTags {
+		if changed, _ := c.updateUpgradeJobs(release, tag); changed {
+			_, err := c.ensureValidationJobs(release, tag)
```

Comment:
This structure unfortunately doesn't work because ensureValidationJobs and ensureAdditionalTests aren't the same thing.  ensureValidationJobs is responsible for changing the phase from Ready to either Accepted or Rejected.  ensureAdditionalTests won't change the phase.  So you should be copying the major loop from ensureValidationJobs into ensureAdditionalTests, removing the phase changes, and making sure errors are returned properly.  Don't try to share this code.